hw/ssi/spi_gpio.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * QEMU model of the SPI GPIO controller
+ *
+ * Copyright (C)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "hw/ssi/spi_gpio.h"
+
+static void cs_line_handler(void *opaque, int n, int level)
+{
+    SpiGpioState *s = SPI_GPIO(opaque);
+    s->cs = !!level;
+}
+
+static void clk_handler(void *opaque, int n, int level)
+{
+    SpiGpioState *s = SPI_GPIO(opaque);
+    s->clk = !!level;
+}
+
+static void mosi_handler(void *opaque, int n, int level)
+{
+    SpiGpioState *s = SPI_GPIO(opaque);
+    s->mosi = !!level;
+}
+
+static void miso_handler(void *opaque, int n, int level)
+{
+    SpiGpioState *s = SPI_GPIO(opaque);
+    s->miso = !!level;
+}
+
+static void spi_gpio_realize(DeviceState *dev, Error **errp)
+{
+    SpiGpioState *s = SPI_GPIO(dev);
+
+    s->spi = ssi_create_bus(dev, "spi");
+
+    qdev_init_gpio_in_named(dev, cs_line_handler, "SPI_CS_IN", 1);
+    qdev_init_gpio_in_named(dev, clk_handler, "SPI_CLK", 1);
+    qdev_init_gpio_in_named(dev, mosi_handler, "SPI_MOSI", 1);
+    qdev_init_gpio_in_named(dev, miso_handler, "SPI_MISO", 1);
```

Comment:
you shall init "SPI_MISO" as output pin use qdev_init_gpio_out_named
Then connect aspeed_gpio's with it. 
```suggestion
    qdev_init_gpio_out_named(dev, s->miso_pin, "SPI_MISO", 1);
```